### PR TITLE
*: Use prometheus for metrics

### PR DESCRIPTION
**Pull request by shenli**
@coocood @qiuyesuifeng @siddontang

### CODE REVIEW

File: util/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		prometheus.HistogramOpts{
+			Namespace: "tidb",
+			Subsystem: "query",
+			Name:      "handle_query_duration_seconds",
+			Help:      "Bucketed histogram of processing time (s) of handled queries.",
+			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 13),
+		})
+)
+
+func QueryMetrics(costTime time.Duration) {
```
Comment (siddontang) [inline util/metrics/metrics.go]: I think Query is ok. because the package is metrics name.

**Comment by shenli** [comment]
PTAL @siddontang

**Comment by shenli** [comment]
vendor updated
PTAL

**Comment by siddontang** [comment]
LGTM

File: server/conn.go [hunk 2]
```diff
@@ -380,7 +381,9 @@ func (cc *clientConn) handleQuery(sql string) (err error) {
 	} else {
 		err = cc.writeOK()
 	}
-	log.Debugf("[TIME_QUERY] %v %s", time.Now().Sub(startTs), sql)
+	costTime := time.Now().Sub(startTs)
+	log.Debugf("[TIME_QUERY] %v %s", costTime, sql)
+	metrics.QueryMetrics(costTime)
```
Comment (qiuyesuifeng) [inline server/conn.go]: How about distinguish whether sql query is succeed or not?

Comment (shenli) [inline server/conn.go]: I will do it in the next PR. This PR is mainly for add prometheus metrics framework.

**Comment by coocood** [comment]
LGTM

**Pull request merged**